### PR TITLE
PKCS #11 Unit Tests

### PR DESCRIPTION
**Pull request by lundinc2**
Description
-----------
* Now at 85% line coverage
* 100% function coverage
* Added unit tests for:
	* C_GetAttributeValue
	* PKCS11_PAL_DESTROY_OBJECT
	* C_Verify
	* C_GenerateKeyPair
	* C_GenerateRandom

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c [hunk 1]
```diff
@@ -287,6 +291,53 @@ static CK_RV prvCreateCert( CK_SESSION_HANDLE_PTR pxSession,
...
+    PKCS11_PAL_FindObject_IgnoreAndReturn( 2 );
+    PKCS11_PAL_GetObjectValue_IgnoreAndReturn( CKR_OK );
+    mbedtls_pk_parse_key_IgnoreAndReturn( 0 );
+    PKCS11_PAL_GetObjectValueCleanup_CMockIgnore();
+    pvPortMalloc_Stub( pvPkcs11MallocCb );
+    mbedtls_ecp_keypair_init_CMockIgnore();
+    mbedtls_ecp_group_init_CMockIgnore();
+    mbedtls_ecp_group_load_IgnoreAndReturn( 0 );
+    mbedtls_mpi_read_binary_IgnoreAndReturn( 0 );
+    pvPortMalloc_Stub( pvPkcs11MallocCb );
```
Comment (alfred2g) [inline libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c:326]: all the stubs are the same
pvPortMalloc_Stub( pvPkcs11MallocCb );

File: libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c [hunk 2]
```diff
@@ -842,14 +927,17 @@ void test_pkcs11_C_CreateObjectECPrivKey( void )
...
+    mbedtls_strerror_highlevel_IgnoreAndReturn( NULL );
+    mbedtls_strerror_lowlevel_IgnoreAndReturn( NULL );
     PKCS11_PAL_GetObjectValueCleanup_CMockIgnore();
-    pvPortMalloc_Stub( pvPkcs11MallocCb );
+    pvPortMalloc_IgnoreAndReturn( &xKeyContext );
     mbedtls_ecp_keypair_init_CMockIgnore();
     mbedtls_ecp_group_init_CMockIgnore();
     mbedtls_ecp_group_load_IgnoreAndReturn( 0 );
-    mbedtls_mpi_read_binary_IgnoreAndReturn( 0 );
     pvPortMalloc_Stub( pvPkcs11MallocCb );
```
Comment (alfred2g) [inline libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c:939]: here too, 
stubs can only be set at the beginning of the function , no need to redefine them before every function call

File: libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c [hunk 3]
```diff
@@ -881,26 +974,25 @@ void test_pkcs11_C_CreateObjectECPubKey( void )
...
     TEST_ASSERT_EQUAL( CKR_OK, xResult );
 
     mbedtls_pk_init_CMockIgnore();
     pvPortMalloc_Stub( pvPkcs11MallocCb );
-    PKCS11_PAL_FindObject_IgnoreAndReturn( 1 );
-    PKCS11_PAL_GetObjectValue_IgnoreAndReturn( CKR_OK );
+    PKCS11_PAL_FindObject_IgnoreAndReturn( CK_INVALID_HANDLE );
     mbedtls_pk_parse_public_key_IgnoreAndReturn( 0 );
     PKCS11_PAL_GetObjectValueCleanup_CMockIgnore();
     pvPortMalloc_Stub( pvPkcs11MallocCb );
```
Comment (alfred2g) [inline libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c:998]: same

File: libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c (hunk 2 above)
Comment (lundinc2) [reply libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c:930]: This one should stay the same as I have different behavior in the malloc stub before this one

Comment (alfred2g) [reply libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c:930]: what do you mean?

Comment (lundinc2) [reply libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c:930]: pvPortMalloc_IgnoreAndReturn( &xKeyContext ); Above this one injects a struct

**Pull request merged**